### PR TITLE
Fix skybox

### PR DESCRIPTION
**Pull request by panxinmiao**
Our Skybox rendering has always had some problems in the past, details can be found in #353.

I noticed that we restated the coordinate system of pygfx in #460. This helps to make some of our logic clearer, 😄 

In this PR, I corrected the error based on the current Skybox rendering logic (different from the way that `Skybox` is regarded as a common world object in #353).

I left some instructions in the code. 

The examples provided in #353 ("[skybox_validate.py](https://github.com/pygfx/pygfx/pull/353/files#diff-f43428565bced2d45c5714e26f96ee161903967950dc4dfae632a424fe0ba384)" and "[validate_skybox_cubemap.py](https://github.com/pygfx/pygfx/pull/353/files#diff-35f79eff8d6d61c820555f9de7c9df25d54df1c36b52a8419a909df1f83f02c1)") can be used to verify the correctness of the results.

### CODE REVIEW

**Comment by panxinmiao** [comment]
> Although the definitions state that the default up vector is +Y, we do want to make it possible to make e.g. up be +Z. This will likely be by changing the up vector of the camera. In a similar way, we should be able to rotate the skybox.

Er, I don't quite understand the specific meaning, 😅 

My understanding is that the up attribute is just a marker, which only specifies which dimension of the object is upward. When the up attribute of an object is changed, it will not produce actual rotation (changing up will not directly change the transform matrix of the object), nor will it change the attitude of the object directly. What really changes the attitude of the object is the "look_at" function, the up attribute works through the "look_at" function. 

If you want the Skybox can be affected by the rotation of the camera, this is already the case in our current implementation. The method of "ndc_to_world_pos" actually contains the rotation information of the camera.

If you want the sky box itself to have the rotation attribute (in theory, it should be so, the sky box should only be unable to translate freely relative to the camera, and other transformations should be free), it is not true now, because we do not regard skybox as a world object, nor use Skybox's transform matrix. The implementation in #353 is precisely because of this consideration. See: https://github.com/pygfx/pygfx/pull/353#discussion_r1063079728

**Comment by almarklein** [comment]
> My understanding is that the up attribute is just a marker, which only specifies which dimension of the object is upward. 

Fair point, maybe it should be a different property, or indeed just use lookat. I think what's best depends to some extend on some moving parts in the cameras.

**Comment by Korijn** [comment]
@almarklein I'll leave it up to you to approve this PR

**Comment by panxinmiao** [comment]
I modified the code. Now Skybox supports arbitrary rotation instead of relying on the `up` property as the "direction" (now it consistent with the behavior of other world objects). 

In this case, I think maybe we don't need to deal with the `up` property of Skybox any more?

**Comment by almarklein** [comment]
That's a nice solution!

Would be nice to have the examples from #353 as well. Could be in a separate PR though. Or we can merge this, and then you can refactor #353 to only add the examples. Whatever works best for you!

**Review by almarklein — APPROVED** [review]
Let's merge this! We can add the examples from https://github.com/pygfx/pygfx/pull/353 later.

File: examples/feature_demo/skybox_rotate.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+camera = gfx.PerspectiveCamera(70)
+
+camera.position.set(0, 4, 20)
+
+controller = gfx.OrbitController(camera.position.clone())
+controller.add_default_event_handlers(renderer, camera)
+
+
+def animate():
+    controller.update_camera(camera)
```
Comment (almarklein) [inline examples/feature_demo/skybox_rotate.py:47]:
```suggestion
```

**Pull request merged**